### DISCUSSION TITLE
Modeling for Where Clause

### DISCUSSION THREAD

**Discussion by ASL-rmarshall** [cdisc-org/analysis-results-standard#139] [Ideas]
A CMAP representation and corresponding LinkML definition for Where Clause definitions that are needed to identify the input records used for each display.

**Comment by ASL-rmarshall** [comment]
Modelling for where clauses is complete and has been used in the modelling of analysis sets and analysis groups.  Each where clause has an id and either a simple condition (dataset/variable/comparator/value(s)) or a compound expression, which is contains a reference to two or more where clauses with a logical operator ("AND" or "OR") to define how the separate where clauses are combined.

I haven't moved this issue to "Done" because (apart from being used for analysis set and analysis group) where clauses have not be linked into the overall model.  

### Comments/questions:
- Are where clauses defined independently and then referenced from within analyses, or are they only defined within the context of an analysis?
- Constraints are missing from the model - there is currently nothing saying that either a simple condition or a compound expression can be defined (see #17 ), and there is nothing to say that compound expression must have 2 or more where clauses.

**Comment by bhavinbusa** [comment]
We are leaving this in because of the outstanding question of the constraints (see above comments from Richard)

**Comment by ASL-rmarshall** [comment]
As of the end of Sprint 6, there are still some issues with the modelling for where clauses:
- Compound expressions for analysis sets and analysis groups can only be constructed by combining references to other already-defined analysis sets/groups.  It would be better if compound expressions could also be defined from scratch (in the same way that data subsets can be).
- Compound expressions for groups do not differentiate between analysis groups and data groups.  It is possible to define a compound express for an analysis group that references defined data groups and vice versa.  This is not ideal because the analysis groups are groups of subjects whereas data groups are groups of data records.

**Comment by drewcdisc** [comment]
Going to transfer this to a Discussion, as we cannot complete this issue all on its own (PBIs need to be independent).

**Comment by drewcdisc** [comment]
@bhavinbusa @ASL-rmarshall  Issue #15 has been converted to Discussion #139
